src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1343,15 +1343,30 @@ private void StopWatchingForExit()
 
         public override string ToString()
         {
-            if (Associated)
+            string result = base.ToString();
+
+            try
             {
-                string processName = ProcessName;
-                if (processName.Length != 0)
+                if (Associated)
                 {
-                    return string.Format(CultureInfo.CurrentCulture, "{0} ({1})", base.ToString(), processName);
+                    _processInfo ??= ProcessManager.GetProcessInfo(_processId, _machineName);
+                    if (_processInfo is not null)
+                    {
+                        string processName = _processInfo.ProcessName;
+                        if (processName.Length != 0)
+                        {
+                            result = $"{result} ({processName})";
+                        }
+                    }
                 }
             }
-            return base.ToString();
+            catch
+            {
+                // Handle cases where a process would exit straight after our checks
+                // and/or make ProcessManager throw an exception.
+            }
```

Comment:
Is this try/catch still needed?  What might throw here if the process exits?

Reply:
@stephentoub `ProcessManager.GetProcessInfo` can still throw an exception, IIRC - that's why I left the `catch` block there